packages/smooth_app/lib/pages/user_preferences_page.dart

```diff
@@ -39,93 +61,60 @@ class UserPreferencesPage extends StatelessWidget {
         productPreferences.getOrderedImportantAttributeIds();
 
     return Scaffold(
-      appBar: AppBar(
-        title: Text(appLocalizations.myPreferences),
-        actions: <Widget>[
-          IconButton(
-            icon: const Icon(Icons.threesixty_outlined),
-            tooltip: 'Check credentials',
-            onPressed: () async {
-              final bool correct =
-                  await UserManagementHelper.checkAndReMountCredentials();
-
-              final SnackBar snackBar = SnackBar(
-                content: Text('It is $correct'),
-                action: SnackBarAction(
-                  label: 'Logout',
-                  onPressed: () async {
-                    UserManagementHelper.logout();
-                  },
-                ),
-              );
-              ScaffoldMessenger.of(context).showSnackBar(snackBar);
-            },
-          ),
-          // TODO(M123-dev): Remove and replace with expandable mock
-          IconButton(
-            icon: const Icon(Icons.supervised_user_circle),
-            tooltip: 'User management',
-            onPressed: () {
-              Navigator.push<Widget>(
-                context,
-                MaterialPageRoute<Widget>(
-                  builder: (BuildContext context) => const LoginPage(),
-                ),
-              );
-            },
-          ),
-
-          IconButton(
-            icon: const Icon(Icons.rotate_left),
-            tooltip: appLocalizations.reset,
-            onPressed: () => _confirmReset(context),
-          ),
-          IconButton(
-            icon: const Icon(Icons.settings),
-            tooltip: appLocalizations.settingsTitle,
-            onPressed: () => Navigator.push<Widget>(
-              context,
-              MaterialPageRoute<Widget>(
-                builder: (BuildContext context) => const ProfilePage(),
-              ),
-            ),
-          ),
-        ],
-      ),
+      appBar: AppBar(title: Text(appLocalizations.myPreferences)),
       body: ListView(
         children: List<Widget>.generate(
-          groups.length,
-          (int index) => _generateGroup(
-            context,
-            groups[index],
-            userPreferences,
-            productPreferences,
-            _reorderAttributes(groups[index], orderedImportantAttributeIds),
-            appLocalizations,
-          ),
+          _getTotalSize(groups, userPreferences),
+          (int index) {
+            for (int groupIndex = 0; groupIndex < 3; groupIndex++) {
+              if (index-- == 0) {
```

Comment:
This syntax works but it's not so easy to understand, I would prefer something like this but feel free to ignore that

``` dart
index--;
if(index == 0){
```

Reply:
There are indeed things to fix in my code.
But here your suggestion is wrong:
```dart
if(index-- == 0){
// do something with index before subtracting 1
}
```
means
```dart
if(index == 0){
  index --;
// do something
} else {
  index--;
}
```